src/transformers/models/bart/modeling_bart.py

```diff
@@ -1473,8 +1473,9 @@ def __init__(self, config: BartConfig, **kwargs):
             config.num_labels,
             config.classifier_dropout,
         )
-        self.model._init_weights(self.classification_head.dense)
-        self.model._init_weights(self.classification_head.out_proj)
+
+        # Initialize weights and apply final processing
+        self.post_init()
```

Comment:
This leftover in BART clashes with the new logic and testing. It is fixed here and in several copies (and in practice does exactly the same since weights are only initialized once).